devices/Mcp7940xx/Mcp79401.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.I2c;
+using Iot.Device.Common;
+
+namespace Iot.Device.Mcp7940xx
+{
+    /// <summary>
+    /// Battery-Backed I2C Real-Time Clock/Calendar with Pre-Programmed EUI-48 MAC ID.
+    /// </summary>
+    public class Mcp79401 : Mcp7940n
+    {
+        /// <summary>
+        /// Lower address of EEPROM memory region.
+        /// </summary>
+        private const byte LowerAddressBoundEEPROM = 0x0;
+
+        /// <summary>
+        /// Upper address of EEPROM memory region.
+        /// </summary>
+        private const byte UpperAddressBoundEEPROM = 0xEF;
+
+        /// <summary>
+        /// EEPROM Address of the EUI Node Address.
+        /// </summary>
+        protected const byte EuiAddress = 0xF0;
+
+        /// <summary>
+        /// Default I2C address for EEPROM.
+        /// </summary>
+        protected const byte EepromI2cAddress = 0b1010_0111;
+
+        /// <summary>
+        /// The underlying I2C device used for accessing the EEPROM address space.
+        /// </summary>
+        protected readonly I2cDevice _EepromI2cDevice;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Mcp79401" /> class.
+        /// </summary>
+        /// <param name="i2cDevice">The I2C device to use for communication.</param>
+        /// <param name="clockSource">The clocks oscillator configuration.</param>
+        /// <exception cref="ArgumentNullException">Thrown when <paramref name="i2cDevice"/> is <c>null</c>.</exception>
+        public Mcp79401(I2cDevice i2cDevice, ClockSource clockSource)
+            : base(i2cDevice, clockSource)
+        {
+            // Setup separate i2cDevice for connecting to devices eeprom address space.
+            I2cConnectionSettings connectionSettings = new I2cConnectionSettings(_I2cDevice.ConnectionSettings.BusId, EepromI2cAddress, _I2cDevice.ConnectionSettings.BusSpeed);
+            _EepromI2cDevice = new I2cDevice(connectionSettings);
```

Comment:
On the devices that have EEPROM, it is accessed using a device address that is different from the clocks address. Basically they glued two devices together is a single package (well, probably not glued, but the end result is the same). The EEPROM address is always the same, so I just handled setting up the I2cDevice with the same bus ID and speed instead of bothering the user with it.